### PR TITLE
fix(ui): show correct commit branch

### PR DESCRIPTION
**Pull request by cloudhead**
Removes the "master" stub.

### CODE REVIEW

File: proxy/src/coco/source.rs [hunk 1]
```diff
@@ -72,6 +72,8 @@ pub struct Commit {
     pub stats: CommitStats,
     /// The changeset introduced by this commit.
     pub diff: diff::Diff,
+    /// The branch this commit belongs to.
+    pub branch: String,
```
Comment (FintanH) [inline proxy/src/coco/source.rs:76]: I would say we could use the `Branch` type defined in this module

File: proxy/src/coco/source.rs [hunk 2]
```diff
@@ -419,12 +421,26 @@ pub fn commit<'repo>(browser: &mut Browser<'repo>, sha1: &str) -> Result<Commit,
...
+    let branches = browser.revision_branches(sha1)?;
+
+    // If a commit figures in more than one branch, there's no real way to know
+    // which branch to show without additional context. So, we choose the first
+    // branch.
+    let branch = branches.first();
+
+    // Known commits always have at least one branch. If this isn't the case, it's a bug.
+    let branch = branch
+        .expect("known commits must be on a branch")
```
Comment (FintanH) [inline proxy/src/coco/source.rs:433]: I'd prefer we added an error variant than `expect`

Comment (cloudhead) [reply proxy/src/coco/source.rs:433]: Is this an error we want to handle though? If a commit doesn't have a branch, I'm not sure what we're supposed to do?

Comment (FintanH) [reply proxy/src/coco/source.rs]: Well, our error variant gets resurfaced as a `warp::Rejection`. It's more to have a structure to the error and we're already in the `Result<_, error::Error>`

**Comment by cloudhead** [comment]
We should have a broader conversation about unwrap vs. Result. I have my own heuristic for deciding which to use, but I bet it's different from yours.

**Comment by FintanH** [comment]
> We should have a broader conversation about unwrap vs. Result. I have my own heuristic for deciding which to use, but I bet it's different from yours.

I pretty much put my thoughts here: https://github.com/radicle-dev/radicle-decisions/blob/master/proposals/0001.md :smile: I think it could be refined, but I generally avoid panic related code. Error documentation is useful to me as a developer too :)

**Comment by FintanH** [comment]
> Hmm there's no mention of panic/unwrap/expect there though, so doesn't answer the question of what to do in cases where an error is unrecoverable..

Well unrecoverable errors would be the `dyn Error` case. But also, I'm pretty sure none of our errors in `error::Error` are recoverable. And in the future we could use it to have the correct error responses. I think in the case of panicing it will always just be `500`.

**Comment by cloudhead** [comment]
A correct error response would only make sense if you expect the error to happen.  In this case the code is saying that this state cannot happen unless there is a bug in the underlying library, or in git, in which case you want to blow up, because an invariant was broken.

**Pull request merged**